dataplane/config_values.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include<common/config.h>
+#include <nlohmann/json.hpp>
+
+#include <cstdint>
+
+struct ConfigValues {
+	uint64_t port_rx_queue_size = 4096;
+	uint64_t port_tx_queue_size = 4096;
+	uint64_t ring_highPriority_size = 64;
+	uint64_t ring_normalPriority_size = 256;
+	uint64_t ring_lowPriority_size = 64;
+	uint64_t ring_toFreePackets_size = 64;
+	uint64_t ring_log_size = 1024;
+	uint64_t fragmentation_size = 1024;
+	uint64_t fragmentation_timeout_first = 32;
+	uint64_t fragmentation_timeout_last = 16;
+	uint64_t fragmentation_packets_per_flow = 64;
+	uint64_t stateful_firewall_tcp_timeout = 120;
+	uint64_t stateful_firewall_tcp_syn_timeout = YANET_CONFIG_STATE_TIMEOUT_DEFAULT;
+	uint64_t stateful_firewall_tcp_syn_ack_timeout = YANET_CONFIG_STATE_TIMEOUT_DEFAULT;
+	uint64_t stateful_firewall_tcp_fin_timeout = YANET_CONFIG_STATE_TIMEOUT_DEFAULT;
+	uint64_t stateful_firewall_udp_timeout = 30;
+	uint64_t stateful_firewall_other_protocols_timeout = 16;
+	uint64_t gc_step = 8;
+	uint64_t sample_gc_step = 512;
+	uint64_t acl_states4_ht_size = YANET_CONFIG_ACL_STATES4_HT_SIZE;
+	uint64_t acl_states6_ht_size = YANET_CONFIG_ACL_STATES6_HT_SIZE;
+	uint64_t master_mempool_size = 8192;
+	uint64_t nat64stateful_states_size = YANET_CONFIG_NAT64STATEFUL_HT_SIZE;
+	uint64_t kernel_interface_queue_size = YANET_CONFIG_KERNEL_INTERFACE_QUEUE_SIZE;
+	uint64_t balancer_state_ht_size = YANET_CONFIG_BALANCER_STATE_HT_SIZE;
+	uint64_t tsc_active_state = YANET_CONFIG_TSC_ACTIVE_STATE;
+	uint64_t balancer_tcp_timeout = YANET_CONFIG_BALANCER_STATE_TIMEOUT_DEFAULT;
+	uint64_t balancer_tcp_syn_timeout = YANET_CONFIG_BALANCER_STATE_TIMEOUT_DEFAULT;
+	uint64_t balancer_tcp_syn_ack_timeout = YANET_CONFIG_BALANCER_STATE_TIMEOUT_DEFAULT;
+	uint64_t balancer_tcp_fin_timeout = YANET_CONFIG_BALANCER_STATE_TIMEOUT_DEFAULT;
+	uint64_t balancer_udp_timeout = YANET_CONFIG_BALANCER_STATE_TIMEOUT_DEFAULT;
+	uint64_t balancer_other_protocols_timeout = YANET_CONFIG_BALANCER_STATE_TIMEOUT_DEFAULT;
+};
+
+inline void from_json(const nlohmann::json& j, ConfigValues& cfg) {
+	#define from_json_if_exists(some_key) \
```

Comment:
This can be the place where we accidentally break API in case of some field renaming. If so - such values from old JSONs will be no longer valid, and we silently ignore them, falling back to default values.
I'd rather sacrifice code bloat here in the name of clarity at this point.